### PR TITLE
Fix cannot find a descendant at path issue

### PR DESCRIPTION
**Pull request by sanghn**
- Attempt to fix sentry issue https://sentry.io/organizations/employment-hero/issues/3076255657/?project=210777&query=is%3Aunresolved+assigned%3Ame&sort=date&statsPeriod=14d 

- Switch to useRef when creating editor, description https://github.com/ianstormtaylor/slate/issues/4081#issue-810494216
- Add key for link custom and minor refactoring for link plugin spec

### CODE REVIEW

File: lib/src/lib.js [hunk 1]
```diff
@@ -62,16 +62,15 @@ const HeroEditor = ({
...
-  );
+  const editorRef = useRef();
+  if (!editorRef.current)
+    editorRef.current = flow(
+      createEditor,
+      withReact,
+      withId(id),
+      ...map(get('enhanceEditor'))(plugins),
+    )();
+  const editor = editorRef.current;
```
Comment (sanghn) [inline lib/src/lib.js:73]: Ref https://github.com/ianstormtaylor/slate/issues/3198

**Review by sangquangnguyen — APPROVED** [review]
Amazing 🚀

**Review by toan2406 — APPROVED** [review]

**Comment by knpwrs** [comment]
Just curious -- did this fix the issue of not being able to find a descendant for you? I'm seeing something similar and it's unclear to me how things can get into such a state that React throws away the memo but keeps the dom around, and would not throw away state in the same circumstances.

**Comment by sanghn** [comment]
@knpwrs Unfortunately, it didn't fix the issue for me

**Comment by suonlight** [comment]
It seems like both `useMemo` and `useRef` do not fix the issue. I just saw the recent updates from https://github.com/ianstormtaylor/slate/pull/4766/files and official docs: https://docs.slatejs.org/libraries/slate-react#withreact-editor-editor
for using `useState` instead

cc @khacluan (FYI: the new fix of race condition of dom and virtual dom did not fix this issue). I already reassign the sentry issue to your team. Please try my new suggestion above ^

**Pull request merged**